etc/test/cpp/unicode.cpp

```diff
@@ -0,0 +1,158 @@
+
+    EXPECT_FALSE( inside( utf8, { Block::MISCELLANEOUS_SYMBOLS } ) );
+    EXPECT_FALSE( inside( utf8, { Block::DINGBATS } ) );
+    EXPECT_FALSE( inside( utf8, { Block::EMOTICONS, Block::ORNAMENTAL_DINGBATS } ) );
+}
+
+TEST( libstdhl_cpp_unicode_range, of_blocks )
+{
+    // GIVEN
+    const std::vector< Block > blocks = {
+        // 0 BMP
+        Block::MISCELLANEOUS_SYMBOLS,
+        Block::DINGBATS,
+
+        // 1 SMP
+        Block::MISCELLANEOUS_SYMBOLS_AND_PICTOGRAPHS,
+        Block::EMOTICONS,
+        Block::ORNAMENTAL_DINGBATS,
+    };
+
+    // WHEN & THEN
+    for( auto block : blocks )
+    {
+        switch( block )
+        {
+            // Plane 0, the Basic Multilingual Plane (BMP)
+            case Block::MISCELLANEOUS_SYMBOLS:
+            {
+                // U+2600..U+26FF | Miscellaneous Symbols | 256 | 256 | Common
+                const auto range = Range::of( block );
+                EXPECT_EQ( range.start(), 0x2600 );
+                EXPECT_EQ( range.end(), 0x26ff );
+                EXPECT_EQ( range.plane(), Plane::BASIC_MULTILINGUAL );
+                break;
+            }
+            case Block::DINGBATS:
+            {
+                // U+2700..U+27BF | Dingbats | 192 | 192 | Common
+                const auto range = Range::of( block );
+                EXPECT_EQ( range.start(), 0x2700 );
+                EXPECT_EQ( range.end(), 0x27bf );
+                EXPECT_EQ( range.plane(), Plane::BASIC_MULTILINGUAL );
+                break;
+            }
+
+            // Plane 1, the Supplementary Multilingual Plane (SMP)
+            case Block::MISCELLANEOUS_SYMBOLS_AND_PICTOGRAPHS:
+            {
+                // U+1F300..U+1F5FF | Miscellaneous Symbols and Pictographs | 768 | 768 | Common
+                const auto range = Range::of( block );
+                EXPECT_EQ( range.start(), 0x1f300 );
+                EXPECT_EQ( range.end(), 0x1f5ff );
+                EXPECT_EQ( range.plane(), Plane::SUPPLEMENTARY_MULTILINGUAL );
+                break;
+            }
+            case Block::EMOTICONS:
+            {
+                // U+1F600..U+1F64F | Emoticons | 80 | 80 | Common
+                const auto range = Range::of( block );
+                EXPECT_EQ( range.start(), 0x1f600 );
+                EXPECT_EQ( range.end(), 0x1f64f );
+                EXPECT_EQ( range.plane(), Plane::SUPPLEMENTARY_MULTILINGUAL );
+                break;
+            }
+            case Block::ORNAMENTAL_DINGBATS:
+            {
+                // U+1F650..U+1F67F | Ornamental Dingbats | 48 | 48 | Common
+                const auto range = Range::of( block );
+                EXPECT_EQ( range.start(), 0x1f650 );
+                EXPECT_EQ( range.end(), 0x1f67f );
+                EXPECT_EQ( range.plane(), Plane::SUPPLEMENTARY_MULTILINGUAL );
+                break;
+            }
+            case Block::TRANSPORT_AND_MAP_SYMBOLS:
```

Comment:
unreachable?

Better use multiple test cases (or parameterized test case) instead of the for loop.
Makes it easier to follow and easier to detect the faulty block in case of an test case error.

Reply:
@emmanuel099 good catch, I've created an issue, which addresses this comment and will fix this problem with the suggested solution in a future PR